shell/platform/darwin/macos/framework/Source/FlutterEmbedderExternalTextureTest.mm

```diff
@@ -59,7 +59,7 @@ - (CVPixelBufferRef)pixelBuffer {
   CVPixelBufferRef pxbuffer = NULL;
   CVReturn status = CVPixelBufferCreate(kCFAllocatorDefault, _width, _width, _pixelFormatType,
                                         (__bridge CFDictionaryRef)options, &pxbuffer);
-  NSAssert(status == kCVReturnSuccess && pxbuffer != NULL, @"Failed to create pixel buffer.");
+  FML_DCHECK(status == kCVReturnSuccess && pxbuffer != nullptr) << "Failed to create pixel buffer";
```

Comment:
"Value stored to 'status' during its initialization is never read"
```
/Volumes/Work/s/w/ir/cache/builder/src/flutter/shell/platform/darwin/macos/framework/Source/FlutterEmbedderExternalTextureTest.mm:60:12: error: Value stored to 'status' during its initialization is never read [clang-analyzer-deadcode.DeadStores,-warnings-as-errors]
   60 |   CVReturn status = CVPixelBufferCreate(kCFAllocatorDefault, _width, _width, _pixelFormatType,
      |            ^~~~~~   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   61 |                                         (__bridge CFDictionaryRef)options, &pxbuffer);
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Hmm, why would this be happening with the `FML_DCHECK` since, as you point out, it shouldn't cause unused variable warnings?
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8746648441899682977/+/u/test:_test:_lint_host_debug/stdout

Reply:
Well, I'll try to run clang-tidy locally to check if there are any problems.

Reply:
I added another comment w.r.t. the advantages of `FML_DCHECK` over `NSAssert` in the context of the Flutter engine codebase. [Link to the comment](https://github.com/flutter/buildroot/pull/860#issuecomment-2138122813)